packages/cli/index.ts

```diff
@@ -365,6 +365,7 @@ async function run() {
       const hooksMap = await retrieveHooks(hooks);
       const meta: VikeMeta = {
         BATI: new Set(flags as Flags[]),
+        BATI_TEST: process.env.NODE_ENV === "test",
```

Comment:
@magne4000,

I've tried playing around with this, and I believe:
- If we rely on `Vitest` to set this environment variable, then it's too early.
- If we explicitly set that environment variable (`NODE_ENV=test`) in a `.env.test` file, than we'll get this wierd error:

```shell
$ pnpm run test:e2e --filter react,fastify,lucia-auth,drizzle

> bati@0.0.242 test:e2e /home/muhammad/Documents/github/phonzammi/vikejs/bati/bati
> pnpm --filter "@batijs/tests" test:e2e "--filter" "react,fastify,lucia-auth,drizzle"


> @batijs/tests@0.0.242 test:e2e /home/muhammad/Documents/github/phonzammi/vikejs/bati/bati/packages/tests
> tsx src/index.ts "--filter" "react,fastify,lucia-auth,drizzle"

file:///home/muhammad/Documents/github/phonzammi/vikejs/bati/bati/node_modules/.pnpm/vitest@2.0.5_@types+node@18.19.43_happy-dom@14.12.3/node_modules/vitest/dist/chunks/utils.Ck2hJTRs.js:5
    throw new Error(errorMsg);
          ^

Error: Vitest failed to access its internal state.

One of the following is possible:
- "vitest" is imported directly without running "vitest" command
- "vitest" is imported inside "globalSetup" (to fix this, use "setupFiles" instead, because "globalSetup" runs in a different context)
- Otherwise, it might be a Vitest bug. Please report it to https://github.com/vitest-dev/vitest/issues

    at getWorkerState (file:///home/muhammad/Documents/github/phonzammi/vikejs/bati/bati/node_modules/.pnpm/vitest@2.0.5_@types+node@18.19.43_happy-dom@14.12.3/node_modules/vitest/dist/chunks/utils.Ck2hJTRs.js:5:11)
    at getTestFile (file:///home/muhammad/Documents/github/phonzammi/vikejs/bati/bati/node_modules/.pnpm/vitest@2.0.5_@types+node@18.19.43_happy-dom@14.12.3/node_modules/vitest/dist/chunks/vi.fiQ7lMRF.js:523:17)
    at createExpect (file:///home/muhammad/Documents/github/phonzammi/vikejs/bati/bati/node_modules/.pnpm/vitest@2.0.5_@types+node@18.19.43_happy-dom@14.12.3/node_modules/vitest/dist/chunks/vi.fiQ7lMRF.js:466:20)
    at file:///home/muhammad/Documents/github/phonzammi/vikejs/bati/bati/node_modules/.pnpm/vitest@2.0.5_@types+node@18.19.43_happy-dom@14.12.3/node_modules/vitest/dist/chunks/vi.fiQ7lMRF.js:526:22
    at ModuleJob.run (node:internal/modules/esm/module_job:222:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at describeBati (/home/muhammad/Documents/github/phonzammi/vikejs/bati/bati/packages/tests-utils/dist/index.js:255:18)
    at <anonymous> (/home/muhammad/Documents/github/phonzammi/vikejs/bati/bati/packages/tests/tests/FRAMEWORK+prisma.spec.ts:5:1)

Node.js v20.16.0
/home/muhammad/Documents/github/phonzammi/vikejs/bati/bati/packages/tests:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @batijs/tests@0.0.242 test:e2e: `tsx src/index.ts "--filter" "react,fastify,lucia-auth,drizzle"`
Exit status 1
 ELIFECYCLE  Command failed with exit code 1.
```

Reply:
Maybe we can use a different environment variable name or value when starting the CLI command, without needing to put it in a `.env.test` file

Reply:
@phonzammi https://github.com/vikejs/bati/commit/449bf8179f542caa37342ab5770cd07689cb42c9 should fix this, but feel free to submit a working workaround if this still doesn't work.

Reply:
Yes, It works now. Thanks